src/preprocess/preprocess.py

```diff
@@ -0,0 +1,113 @@
+
+
+def generate_queries(departament, f):
+    for key in departament.keys():
+        f.write("--"+key + "\n")
+        if "actions" not in departament[key]:
+            continue
+        for rule in departament[key]["actions"]:
+            has_datasets = False
+            has_event_program = False
+            has_tracker_program = False
+            if rule["action"] == remove_rule:
+                datasets = ""
+                event_program = ""
+                tracker_program = ""
+                if "selectMDType" in rule.keys():
+                    for type in rule["selectMDType"]:
+                        if type.lower() == dataset_type:
+                            has_datasets = True
+                        if type.lower() == program_type:
+                            has_event_program = True
+                        if type.lower() == tracker_type:
+                            has_tracker_program = True
+                        if "selectDatasets" in rule.keys():
+                            datasets = rule["selectDatasets"]
+                            if dataset_type not in type:
+                                has_datasets = True
+                        if "selectEventProgram" in rule.keys():
+                            event_program = rule["selectEventProgram"]
+                            if program_type not in type:
+                                has_event_program = True
+                        if "selectTrackerProgram" in rule.keys():
+                            tracker_program = rule["selectTrackerProgram"]
+                            if tracker_program not in type:
+                                has_tracker_program = True
+
+                org_units = ""
+                if "selectOrgUnits" in rule.keys():
+                    org_units = rule["selectOrgUnits"]
+                org_unit_descendants = ""
+                if "selectOrgUnitAndDescendants" in rule.keys():
+                    org_unit_descendants = rule["selectOrgUnitAndDescendants"]
+                data_elements = ""
+                if "selectDataElements" in rule.keys():
+                    data_elements = rule["selectDataElements"]
+
+                if has_datasets:
+                    generate_delete_datasets_rules(datasets, data_elements, org_units, org_unit_descendants, departament[key]["datasets"], f)
+                if has_tracker_program:
+                    generate_delete_tracker_rules(tracker_program, data_elements, org_units, org_unit_descendants, departament[key]["trackerprograms"], f)
+                if has_event_program:
+                    generate_delete_event_rules(event_program, data_elements, org_units, org_unit_descendants, departament[key]["eventprograms"], f)
+                if not has_datasets and not has_event_program and not has_tracker_program:
+                    remove_all(departament[key], f)
+
+
+def add_rules_by_departament(departament, entries):
+    for entry in [x for x in entries]:
```

Comment:
Maybe we were doing some transformation before, but as it's now: `for entry in entries:`